openblas_utest_ext tests are failing on s390x with Z13 kernel

I am seeing test failures after the introduction of the `openblas_utest_ext` tests (commit bf39c0d8b53c8899fb81a9515da91a004d25ec0b) on s390x with `Z13` kernel (`ZARCH_GENERIC` and `Z14` are OK) when building on Fedora 38 (`gcc-13.2.1-4.fc38.s390x`).
````
...
TEST 203/1542 izamin:step_zero [OK]
TEST 204/1542 izamin:bad_args_N_0 [OK]
TEST 205/1542 ssum:c_api_step_2_N_50 [OK]
TEST 206/1542 ssum:c_api_step_1_N_50 [FAIL]
  ERR: test_extensions/test_ssum.c:386  expected 0.000e+00, got 5.000e+01 (diff -5.000e+01, tol 1.000e-04)
TEST 207/1542 ssum:c_api_step_2_N_5 [OK]
TEST 208/1542 ssum:c_api_step_1_N_5 [OK]
TEST 209/1542 ssum:c_api_step_2_N_4 [FAIL]
  ERR: test_extensions/test_ssum.c:347  expected 1.000e+00, got 7.600e+00 (diff -6.600e+00, tol 1.000e-04)
TEST 210/1542 ssum:c_api_step_1_N_4 [FAIL]
  ERR: test_extensions/test_ssum.c:335  expected 3.200e+00, got 7.600e+00 (diff -4.400e+00, tol 1.000e-04)
TEST 211/1542 ssum:c_api_step_2_N_3 [FAIL]
  ERR: test_extensions/test_ssum.c:323  expected 2.300e+00, got 4.300e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 212/1542 ssum:c_api_step_1_N_3 [OK]
TEST 213/1542 ssum:c_api_step_2_N_2 [OK]
TEST 214/1542 ssum:c_api_step_1_N_2 [FAIL]
  ERR: test_extensions/test_ssum.c:287  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 215/1542 ssum:c_api_step_2_N_1 [OK]
TEST 216/1542 ssum:c_api_step_1_N_1 [OK]
TEST 217/1542 ssum:c_api_step_zero [OK]
TEST 218/1542 ssum:c_api_bad_args_N_0 [OK]
TEST 219/1542 ssum:step_2_N_50 [OK]
TEST 220/1542 ssum:step_1_N_50 [FAIL]
  ERR: test_extensions/test_ssum.c:205  expected 0.000e+00, got 5.000e+01 (diff -5.000e+01, tol 1.000e-04)
TEST 221/1542 ssum:step_2_N_5 [OK]
TEST 222/1542 ssum:step_1_N_5 [OK]
TEST 223/1542 ssum:step_2_N_4 [FAIL]
  ERR: test_extensions/test_ssum.c:166  expected 1.000e+00, got 7.600e+00 (diff -6.600e+00, tol 1.000e-04)
TEST 224/1542 ssum:step_1_N_4 [FAIL]
  ERR: test_extensions/test_ssum.c:154  expected 3.200e+00, got 7.600e+00 (diff -4.400e+00, tol 1.000e-04)
TEST 225/1542 ssum:step_2_N_3 [FAIL]
  ERR: test_extensions/test_ssum.c:142  expected 2.300e+00, got 4.300e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 226/1542 ssum:step_1_N_3 [OK]
TEST 227/1542 ssum:step_2_N_2 [OK]
TEST 228/1542 ssum:step_1_N_2 [FAIL]
  ERR: test_extensions/test_ssum.c:106  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 229/1542 ssum:step_2_N_1 [OK]
TEST 230/1542 ssum:step_1_N_1 [OK]
TEST 231/1542 ssum:step_zero [OK]
TEST 232/1542 ssum:bad_args_N_0 [OK]
TEST 233/1542 dsum:c_api_step_2_N_50 [OK]
TEST 234/1542 dsum:c_api_step_1_N_50 [FAIL]
  ERR: test_extensions/test_dsum.c:386  expected 0.000e+00, got 5.000e+01 (diff -5.000e+01, tol 1.000e-13)
TEST 235/1542 dsum:c_api_step_2_N_5 [OK]
TEST 236/1542 dsum:c_api_step_1_N_5 [OK]
TEST 237/1542 dsum:c_api_step_2_N_4 [FAIL]
  ERR: test_extensions/test_dsum.c:347  expected 1.000e+00, got 7.600e+00 (diff -6.600e+00, tol 1.000e-13)
TEST 238/1542 dsum:c_api_step_1_N_4 [FAIL]
  ERR: test_extensions/test_dsum.c:335  expected 3.200e+00, got 7.600e+00 (diff -4.400e+00, tol 1.000e-13)
TEST 239/1542 dsum:c_api_step_2_N_3 [FAIL]
  ERR: test_extensions/test_dsum.c:323  expected 2.300e+00, got 4.300e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 240/1542 dsum:c_api_step_1_N_3 [OK]
TEST 241/1542 dsum:c_api_step_2_N_2 [OK]
TEST 242/1542 dsum:c_api_step_1_N_2 [FAIL]
  ERR: test_extensions/test_dsum.c:287  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 243/1542 dsum:c_api_step_2_N_1 [OK]
TEST 244/1542 dsum:c_api_step_1_N_1 [OK]
TEST 245/1542 dsum:c_api_step_zero [OK]
TEST 246/1542 dsum:c_api_bad_args_N_0 [OK]
TEST 247/1542 dsum:step_2_N_50 [OK]
TEST 248/1542 dsum:step_1_N_50 [FAIL]
  ERR: test_extensions/test_dsum.c:205  expected 0.000e+00, got 5.000e+01 (diff -5.000e+01, tol 1.000e-13)
TEST 249/1542 dsum:step_2_N_5 [OK]
TEST 250/1542 dsum:step_1_N_5 [OK]
TEST 251/1542 dsum:step_2_N_4 [FAIL]
  ERR: test_extensions/test_dsum.c:166  expected 1.000e+00, got 7.600e+00 (diff -6.600e+00, tol 1.000e-13)
TEST 252/1542 dsum:step_1_N_4 [FAIL]
  ERR: test_extensions/test_dsum.c:154  expected 3.200e+00, got 7.600e+00 (diff -4.400e+00, tol 1.000e-13)
TEST 253/1542 dsum:step_2_N_3 [FAIL]
  ERR: test_extensions/test_dsum.c:142  expected 2.300e+00, got 4.300e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 254/1542 dsum:step_1_N_3 [OK]
TEST 255/1542 dsum:step_2_N_2 [OK]
TEST 256/1542 dsum:step_1_N_2 [FAIL]
  ERR: test_extensions/test_dsum.c:106  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 257/1542 dsum:step_2_N_1 [OK]
TEST 258/1542 dsum:step_1_N_1 [OK]
TEST 259/1542 dsum:step_zero [OK]
TEST 260/1542 dsum:bad_args_N_0 [OK]
TEST 261/1542 scsum:c_api_step_2_N_50 [FAIL]
  ERR: test_extensions/test_scsum.c:401  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-04)
TEST 262/1542 scsum:c_api_step_1_N_50 [FAIL]
  ERR: test_extensions/test_scsum.c:386  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-04)
TEST 263/1542 scsum:c_api_step_2_N_5 [FAIL]
  ERR: test_extensions/test_scsum.c:371  expected 1.110e+01, got 3.690e+01 (diff -2.580e+01, tol 1.000e-04)
TEST 264/1542 scsum:c_api_step_1_N_5 [OK]
TEST 265/1542 scsum:c_api_step_2_N_4 [FAIL]
  ERR: test_extensions/test_scsum.c:347  expected -2.000e-01, got 1.820e+01 (diff -1.840e+01, tol 1.000e-04)
TEST 266/1542 scsum:c_api_step_1_N_4 [FAIL]
  ERR: test_extensions/test_scsum.c:335  expected 6.400e+00, got 1.520e+01 (diff -8.800e+00, tol 1.000e-04)
TEST 267/1542 scsum:c_api_step_2_N_3 [FAIL]
  ERR: test_extensions/test_scsum.c:323  expected 2.300e+00, got 1.030e+01 (diff -8.000e+00, tol 1.000e-04)
TEST 268/1542 scsum:c_api_step_1_N_3 [FAIL]
  ERR: test_extensions/test_scsum.c:311  expected 4.400e+00, got 6.400e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 269/1542 scsum:c_api_step_2_N_2 [FAIL]
  ERR: test_extensions/test_scsum.c:299  expected 1.600e+00, got 4.600e+00 (diff -3.000e+00, tol 1.000e-04)
TEST 270/1542 scsum:c_api_step_1_N_2 [FAIL]
  ERR: test_extensions/test_scsum.c:287  expected 1.400e+00, got 5.400e+00 (diff -4.000e+00, tol 1.000e-04)
TEST 271/1542 scsum:c_api_step_2_N_1 [OK]
TEST 272/1542 scsum:c_api_step_1_N_1 [FAIL]
  ERR: test_extensions/test_scsum.c:263  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 273/1542 scsum:c_api_step_zero [OK]
TEST 274/1542 scsum:c_api_bad_args_N_0 [OK]
TEST 275/1542 scsum:step_2_N_50 [FAIL]
  ERR: test_extensions/test_scsum.c:220  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-04)
TEST 276/1542 scsum:step_1_N_50 [FAIL]
  ERR: test_extensions/test_scsum.c:205  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-04)
TEST 277/1542 scsum:step_2_N_5 [FAIL]
  ERR: test_extensions/test_scsum.c:190  expected 1.110e+01, got 3.690e+01 (diff -2.580e+01, tol 1.000e-04)
TEST 278/1542 scsum:step_1_N_5 [OK]
TEST 279/1542 scsum:step_2_N_4 [FAIL]
  ERR: test_extensions/test_scsum.c:166  expected -2.000e-01, got 1.820e+01 (diff -1.840e+01, tol 1.000e-04)
TEST 280/1542 scsum:step_1_N_4 [FAIL]
  ERR: test_extensions/test_scsum.c:154  expected 6.400e+00, got 1.520e+01 (diff -8.800e+00, tol 1.000e-04)
TEST 281/1542 scsum:step_2_N_3 [FAIL]
  ERR: test_extensions/test_scsum.c:142  expected 2.300e+00, got 1.030e+01 (diff -8.000e+00, tol 1.000e-04)
TEST 282/1542 scsum:step_1_N_3 [FAIL]
  ERR: test_extensions/test_scsum.c:130  expected 4.400e+00, got 6.400e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 283/1542 scsum:step_2_N_2 [FAIL]
  ERR: test_extensions/test_scsum.c:118  expected 1.600e+00, got 4.600e+00 (diff -3.000e+00, tol 1.000e-04)
TEST 284/1542 scsum:step_1_N_2 [FAIL]
  ERR: test_extensions/test_scsum.c:106  expected 1.400e+00, got 5.400e+00 (diff -4.000e+00, tol 1.000e-04)
TEST 285/1542 scsum:step_2_N_1 [OK]
TEST 286/1542 scsum:step_1_N_1 [FAIL]
  ERR: test_extensions/test_scsum.c:82  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-04)
TEST 287/1542 scsum:step_zero [OK]
TEST 288/1542 scsum:bad_args_N_0 [OK]
TEST 289/1542 dzsum:c_api_step_2_N_50 [FAIL]
  ERR: test_extensions/test_dzsum.c:401  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-13)
TEST 290/1542 dzsum:c_api_step_1_N_50 [FAIL]
  ERR: test_extensions/test_dzsum.c:386  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-13)
TEST 291/1542 dzsum:c_api_step_2_N_5 [FAIL]
  ERR: test_extensions/test_dzsum.c:371  expected 1.110e+01, got 3.690e+01 (diff -2.580e+01, tol 1.000e-13)
TEST 292/1542 dzsum:c_api_step_1_N_5 [OK]
TEST 293/1542 dzsum:c_api_step_2_N_4 [FAIL]
  ERR: test_extensions/test_dzsum.c:347  expected -2.000e-01, got 1.820e+01 (diff -1.840e+01, tol 1.000e-13)
TEST 294/1542 dzsum:c_api_step_1_N_4 [FAIL]
  ERR: test_extensions/test_dzsum.c:335  expected 6.400e+00, got 1.520e+01 (diff -8.800e+00, tol 1.000e-13)
TEST 295/1542 dzsum:c_api_step_2_N_3 [FAIL]
  ERR: test_extensions/test_dzsum.c:323  expected 2.300e+00, got 1.030e+01 (diff -8.000e+00, tol 1.000e-13)
TEST 296/1542 dzsum:c_api_step_1_N_3 [FAIL]
  ERR: test_extensions/test_dzsum.c:311  expected 4.400e+00, got 6.400e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 297/1542 dzsum:c_api_step_2_N_2 [FAIL]
  ERR: test_extensions/test_dzsum.c:299  expected 1.600e+00, got 4.600e+00 (diff -3.000e+00, tol 1.000e-13)
TEST 298/1542 dzsum:c_api_step_1_N_2 [FAIL]
  ERR: test_extensions/test_dzsum.c:287  expected 1.400e+00, got 5.400e+00 (diff -4.000e+00, tol 1.000e-13)
TEST 299/1542 dzsum:c_api_step_2_N_1 [OK]
TEST 300/1542 dzsum:c_api_step_1_N_1 [FAIL]
  ERR: test_extensions/test_dzsum.c:263  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 301/1542 dzsum:c_api_step_zero [OK]
TEST 302/1542 dzsum:c_api_bad_args_N_0 [OK]
TEST 303/1542 dzsum:step_2_N_50 [FAIL]
  ERR: test_extensions/test_dzsum.c:220  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-13)
TEST 304/1542 dzsum:step_1_N_50 [FAIL]
  ERR: test_extensions/test_dzsum.c:205  expected 0.000e+00, got 1.000e+02 (diff -1.000e+02, tol 1.000e-13)
TEST 305/1542 dzsum:step_2_N_5 [FAIL]
  ERR: test_extensions/test_dzsum.c:190  expected 1.110e+01, got 3.690e+01 (diff -2.580e+01, tol 1.000e-13)
TEST 306/1542 dzsum:step_1_N_5 [OK]
TEST 307/1542 dzsum:step_2_N_4 [FAIL]
  ERR: test_extensions/test_dzsum.c:166  expected -2.000e-01, got 1.820e+01 (diff -1.840e+01, tol 1.000e-13)
TEST 308/1542 dzsum:step_1_N_4 [FAIL]
  ERR: test_extensions/test_dzsum.c:154  expected 6.400e+00, got 1.520e+01 (diff -8.800e+00, tol 1.000e-13)
TEST 309/1542 dzsum:step_2_N_3 [FAIL]
  ERR: test_extensions/test_dzsum.c:142  expected 2.300e+00, got 1.030e+01 (diff -8.000e+00, tol 1.000e-13)
TEST 310/1542 dzsum:step_1_N_3 [FAIL]
  ERR: test_extensions/test_dzsum.c:130  expected 4.400e+00, got 6.400e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 311/1542 dzsum:step_2_N_2 [FAIL]
  ERR: test_extensions/test_dzsum.c:118  expected 1.600e+00, got 4.600e+00 (diff -3.000e+00, tol 1.000e-13)
TEST 312/1542 dzsum:step_1_N_2 [FAIL]
  ERR: test_extensions/test_dzsum.c:106  expected 1.400e+00, got 5.400e+00 (diff -4.000e+00, tol 1.000e-13)
TEST 313/1542 dzsum:step_2_N_1 [OK]
TEST 314/1542 dzsum:step_1_N_1 [FAIL]
  ERR: test_extensions/test_dzsum.c:82  expected 1.000e-01, got 2.100e+00 (diff -2.000e+00, tol 1.000e-13)
TEST 315/1542 dzsum:step_zero [OK]
TEST 316/1542 dzsum:bad_args_N_0 [OK]
TEST 317/1542 saxpby:c_api_check_n_zero [OK]
...
TEST 883/1542 cimatcopy:rowmajor_conjtrans_col_50_row_100 [OK]
TEST 884/1542 cimatcopy:rowmajor_conj_col_50_row_100 double free or corruption (!prev)
make[1]: *** [Makefile:73: run_test] Aborted (core dumped)
````

The `abort` in test `884` goes away in some later commit (it's not present when building HEAD)

--- a/kernel/zarch/KERNEL.Z13
+++ b/kernel/zarch/KERNEL.Z13
@@ -35,10 +35,10 @@ DASUMKERNEL  = dasum.c
 CASUMKERNEL  = ../arm/zasum.c
 ZASUMKERNEL  = zasum.c
 
-SSUMKERNEL  = ../arm/asum.c
-DSUMKERNEL  = dasum.c
-CSUMKERNEL  = ../arm/zasum.c
-ZSUMKERNEL  = zasum.c
+SSUMKERNEL  = ../arm/sum.c
+DSUMKERNEL  = dsum.c
+CSUMKERNEL  = ../arm/zsum.c
+ZSUMKERNEL  = zsum.c
 
 SAXPYKERNEL  = ../arm/axpy.c
 DAXPYKERNEL  = daxpy.c

